core/src/main/java/com/adobe/aem/commons/assetshare/components/search/impl/SearchConfigImpl.java

```diff
@@ -0,0 +1,151 @@
+package com.adobe.aem.commons.assetshare.components.search.impl;
+
+import com.adobe.aem.commons.assetshare.components.search.SearchConfig;
+import com.adobe.aem.commons.assetshare.util.ResourceTypeVisitor;
+import com.day.cq.dam.api.DamConstants;
+import com.day.cq.search.Predicate;
+import com.day.cq.wcm.api.Page;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.models.annotations.DefaultInjectionStrategy;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.OSGiService;
+import org.apache.sling.models.annotations.injectorspecific.ScriptVariable;
+import org.apache.sling.models.annotations.injectorspecific.Self;
+import org.apache.sling.models.factory.ModelFactory;
+
+import javax.annotation.PostConstruct;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Model(
+        adaptables = {SlingHttpServletRequest.class,Resource.class},
+        adapters = {SearchConfig.class},
+        defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL
```

Comment:
This is triggering errors in the log file:
```
23.10.2018 23:27:11.008 *WARN* [52.86.210.48 [1540337173637] GET /content/portals/global/en/resource-library.results.html HTTP/1.1] org.apache.sling.models.impl.ModelAdapterFactory Although falling back to default value worked, injection into private org.apache.sling.api.resource.Resource com.adobe.aem.commons.assetshare.components.search.impl.SearchConfigImpl.resource failed because of: Could not adapt from class org.apache.sling.engine.impl.SlingHttpServletRequestImpl to interface org.apache.sling.api.resource.Resource
org.apache.sling.models.factory.ModelClassException: Could not adapt from class org.apache.sling.engine.impl.SlingHttpServletRequestImpl to interface org.apache.sling.api.resource.Resource
```

I wonder if instead of injecting `resource` with `@Self` (if that's truly the issue) if we can instead calculate `resource` from the request inside of the `init()` function.

Reply:
I addressed this error.